modules/javafx.graphics/src/main/java/com/sun/javafx/font/directwrite/DWGlyphLayout.java

```diff
@@ -138,14 +138,15 @@ public void layout(TextRun run, PGFont font, FontStrike strike, char[] text) {
         int i, j;
         int[] iglyphs = new int[glyphCount];
         int slotMask = slot << 24;
+        final int SHORTMASK = 0x0000ffff;
         boolean missingGlyph = false;
         i = 0; j = rtl ? glyphCount - 1 : 0;
         while (i < glyphCount) {
             if (glyphs[i] == 0) {
                 missingGlyph = true;
                 if (composite) break;
             }
-            iglyphs[i] = glyphs[j] | slotMask;
+            iglyphs[i] = ((int)glyphs[j] & SHORTMASK) | slotMask;
```

Comment:
The cast to `(int)` is redundant: `glyphs[j]` is an int.

Reply:
glyphs[j] is short.
I now checked following code works.
iglyphs[i] = (glyphs[j] & SHORTMASK) | slotMask;

Reply:
You're right. I was looking at iglyphs[]. With a short, the cast is optional, so it's up to you whether to keep it.